### DISCUSSION TITLE
Progress when copying image with rauc adaptive updates

### DISCUSSION THREAD

**Discussion by damien-projectx** [rauc/rauc#1564] [Q&A]
We have rauc configured to perform adaptive updates:

- Bundle format: "verity"
- Slot filesystem: "ext4"
- Slot adaptive: "block-hash-index"

During the update the "copy" step can take a long time but does not show any progress:

```
 46% Copying image to rootfs.0
 99% Copying image to rootfs.0 done.
```

Is this a missing feature of rauc or did we miss-configure it?

**Comment by jluebbe** [comment]
It's a missing feature.

Ideally, we'd check which blocks need to be downloaded, but the simple way is to add integration to `r_context_set_step_percentage()` to the [loop in copy_block_hash_index_to_dev()](https://github.com/rauc/rauc/blob/master/src/update_handler.c#L793), similar to [splice_with_progress()](https://github.com/rauc/rauc/blob/master/src/update_handler.c#L184).

**Comment by ejoerns** [comment]
Duplicate of #1277
